WordPress/Classes/ViewRelated/Menus/Views/MenuItemSourceFooterView.m

```diff
@@ -0,0 +1,196 @@
+
+@end
+
+@interface MenuItemSourceLoadingDrawView : UIView
+
+@property (nonatomic, weak) id <MenuItemSourceLoadingDrawViewDelegate> drawDelegate;
+
+@end
+
+@interface MenuItemSourceFooterView () <MenuItemSourceLoadingDrawViewDelegate>
+
+@property (nonatomic, copy) NSString *labelText;
+@property (nonatomic, assign) BOOL drawsLabelTextIfNeeded;
+@property (nonatomic, strong) MenuItemSourceCell *sourceCell;
+@property (nonatomic, strong) MenuItemSourceLoadingDrawView *drawView;
+@property (nonatomic, strong) NSTimer *beginLoadingAnimationsTimer;
+@property (nonatomic, strong) NSTimer *endLoadingAnimationsTimer;
+
+@end
+
+@implementation MenuItemSourceFooterView
+
+- (void)dealloc
+{
+    [self.beginLoadingAnimationsTimer invalidate];
+    [self.endLoadingAnimationsTimer invalidate];
+}
+
+- (id)initWithFrame:(CGRect)frame
+{
+    self = [super initWithFrame:frame];
+    if (self) {
+        
+        self.backgroundColor = [UIColor whiteColor];
+        
+        [self initSourceCell];
+        [self initDrawView];
+    }
+    
+    return self;
+}
+
+- (void)initSourceCell
+{
+    MenuItemSourceCell *cell = [[MenuItemSourceCell alloc] initWithStyle:UITableViewCellStyleDefault reuseIdentifier:nil];
+    cell.frame = self.bounds;
+    cell.autoresizingMask = UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight;
+    cell.alpha = 0.0;
+    [cell setTitle:@"Dummy Text For Sizing the Label"];
+    [self addSubview:cell];
+    self.sourceCell = cell;
+}
+
+- (void)initDrawView
+{
+    MenuItemSourceLoadingDrawView *drawView = [[MenuItemSourceLoadingDrawView alloc] initWithFrame:self.bounds];
+    drawView.backgroundColor = [UIColor whiteColor];
+    drawView.autoresizingMask = UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight;
+    drawView.drawDelegate = self;
+    drawView.contentMode = UIViewContentModeRedraw;
+    [self.sourceCell addSubview:drawView];
+    self.drawView = drawView;
+}
+
+- (void)toggleMessageWithText:(NSString *)text
+{
+    self.labelText = text;
+    if (!self.beginLoadingAnimationsTimer && !self.endLoadingAnimationsTimer) {
+        self.drawsLabelTextIfNeeded = YES;
+    }
+}
+
+- (void)startLoadingIndicatorAnimation
+{
+    if (self.isAnimating) {
+        return;
+    }
+    
+    self.drawsLabelTextIfNeeded = NO;
+    
+    [self.beginLoadingAnimationsTimer invalidate];
+    self.beginLoadingAnimationsTimer = nil;
+    [self.endLoadingAnimationsTimer invalidate];
+    self.endLoadingAnimationsTimer = nil;
+    
+    self.isAnimating = YES;
+    self.sourceCell.hidden = NO;
+    
+    // Will begin animations on next runloop incase there are upcoming layout upates in-which the animation won't play.
+    NSTimer *timer = [NSTimer timerWithTimeInterval:0.0 target:self selector:@selector(beginCellAnimations) userInfo:nil repeats:NO];
+    self.beginLoadingAnimationsTimer = timer;
+    // Add the timer to the runloop scheduling under common modes, to not pause for UIScrollView scrolling.
```

Comment:
Changing to `under common modes so it does not pause while a UIScrollView scrolls.`